src/v/cluster/data_migration_group_proxy.h

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#pragma once
+
+#include "base/outcome.h"
+#include "cluster/errc.h"
+#include "container/fragmented_vector.h"
+#include "kafka/protocol/types.h"
+#include "model/fundamental.h"
+#include "model/timeout_clock.h"
+
+namespace cluster::data_migrations {
+
+/*
+ * Abstract class for a proxy component to operate kafka layer consumer groups
+ * from data migration components.
+ */
+class group_proxy {
```

Comment:
i am wondering if `group_proxy` could leverage the pimpl itself, the the `std::unique_ptr` trick wouldn't be required in sharded but rather than that we would use a `make_group_proxy` method

Reply:
As discussed on slack, I actually prefer this "trick": why create our own wrapper class if `unique_ptr` does exactly what we need? In particular, I like the idea of passing a reference to implementation itself to dependent components, to avoid extra indirections when calling. That's to improve both readability and performance.